### PR TITLE
Remove calls to deprecated `Tensor.storage()` when using newer PyTorch versions

### PR DESCRIPTION
**Pull request by mrfh92**
Intended to resolve issue #1229

### CODE REVIEW

**Comment by mrfh92** [comment]
merged `main` (although this is labeled as "bug"-PR) because it is actually not a bug-fix but rather a reaction to some future development in PyTorch and therefore this PR can wait til next release to become "proper" part of Heat

**Comment by mrfh92** [comment]
The first idea for a workaround mentioned above that resulted in errors only in `test_stride_and_strides` does not seem to work for certain earlier PyTorch versions. I have tested with torch==2.0.0 on my worksation (result above), but torch==1.8.1 (also on my workstation) and torch==1.12.0 (on HDFML) which resulted in multiple errors over many unittests.  When I use torch==2.0.0 on HDFML, I only get errors in `test_stride_and_strides` again...

**Comment by mrfh92** [comment]
Still a problem: 
```
/p/project/haf/users/hoppe6/heat/heat/optim/dp_optimizer.py:23: UserWarning: TypedStorage is deprecated. It will be removed in the future and UntypedStorage will be the only storage class. This should only matter to you if you are using storages directly.  To access UntypedStorage directly, use tensor.untyped_storage() instead of tensor.storage()
  tens_a = torch.HalfTensor().set_(torch.HalfStorage.from_buffer(buffer_a, "native"))
```

**Comment by mrfh92** [comment]
Current workaround works on HDFML with torch==1.12 and torch==2.0.0 execpt for the following problems: 

```
/p/project/haf/users/hoppe6/heat/heat/optim/dp_optimizer.py:23: UserWarning: TypedStorage is deprecated. It will be removed in the future and UntypedStorage will be the only storage class. This should only matter to you if you are using storages directly.  To access UntypedStorage directly, use tensor.untyped_storage() instead of tensor.storage()
  tens_a = torch.HalfTensor().set_(torch.HalfStorage.from_buffer(buffer_a, "native"))
```
and similar for lines 24, 33, and 34 of the same file.

**Comment by mrfh92** [comment]
**Comment on decreased codecov:** 
This is mainly due to the fact that I now skipp all DASO-related tests as long as the hardware is inappropriate (i.e., no GPUs or less than 2 nodes), which is the case for our CI-runners. Hence, DASO is not covered anymore.

**Comment by mrfh92** [comment]
Workaround for DASO has successfully been tested on HDFML (2 nodes, 4 GPUs each) with PyTorch==1.12.0 and PyTorch==2.0.0.

File: heat/core/version.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 """Indicates feature extension."""
 micro: int = 0
 """Indicates revisions for bugfixes."""
-extension: str = "dev"
+extension: str = None
```
[the remark this answers is not in the document]
Comment (ClaudiaComito) [reply heat/core/version.py:10]: we're merging into `main`, correct?

**Review by ClaudiaComito — CHANGES_REQUESTED** [review]
Great @mrfh92 thanks a lot for fixing this. I only have one potential change in the `version.py` file since we are merging into `main`.

Thanks a lot!

**Review by ClaudiaComito — APPROVED** [review]
Awesome!

**Pull request merged**